src/databricks/sql/utils.py

```diff
@@ -560,6 +588,29 @@ def convert_decimals_in_arrow_table(table, description) -> pyarrow.Table:
     return table
 
 
+def convert_to_assigned_datatypes_in_column_table(column_table, description):
+    for i, col in enumerate(column_table):
```

Comment:
I would recommend not change `column_table`, instead, return a table with new assigned data types, input argument should be immutable, if the function crashes in the middle, the state of column_table becomes inconsisent.

Reply:
Fixed it